crates/ruff_python_semantic/src/model.rs

```diff
@@ -257,6 +257,27 @@ impl<'a> SemanticModel<'a> {
             .is_some_and(|binding| binding.kind.is_builtin())
     }
 
+    /// Return `true` if `member` is a reference to `builtins.$target`,
+    /// i.e. either `object` (where `object` is not overridden in the global scope),
+    /// or `builtins.object` (where `builtins` is imported as a module at the top level)
+    pub fn match_builtin_expr(&self, expr: &Expr, symbol: &str) -> bool {
+        debug_assert!(!symbol.contains('.'));
+        if self.seen_module(Modules::BUILTINS) {
+            // slow path
+            self.resolve_qualified_name(expr)
+                .is_some_and(|qualified_name| match qualified_name.segments() {
+                    ["builtins" | "", member] => *member == symbol,
+                    _ => false,
+                })
+        } else {
+            // fast(er) path
+            match expr {
+                Expr::Name(ast::ExprName { id, .. }) => id == symbol && self.is_builtin(symbol),
+                _ => false,
+            }
+        }
```

Comment:
Nit, you could use `matches!` here
```suggestion
            matches!(expr, Expr::Name(ast::ExprName { id, .. }) if id == symbol && self.is_builtin(symbol))
        }
```

Reply:
TIL :-)